projects/packages/jetpack-mu-wpcom/src/features/wpcom-comments/class-wp-rest-comment-like.php

```diff
@@ -0,0 +1,147 @@
+<?php
+/**
+ * File: class-wp-rest-comments-likes.php
+ *
+ * Provides REST API endpoints for creating and deleting comment likes.
+ *
+ * @package automattic/jetpack-mu-wpcom
+ */
+
+/**
+ * Class WP_REST_Comments_Likes.
+ *
+ * Handles endpoints for creating a new like and deleting a like.
+ */
+class WP_REST_Comment_Like extends WP_REST_Controller {
+
+	/**
+	 * Constructor.
+	 */
+	public function __construct() {
+		$this->namespace = 'rest/v1.1';
+	}
+
+	/**
+	 * Register available routes.
+	 */
+	public function register_routes() {
+		// Endpoint for creating a new like.
+		register_rest_route(
+			$this->namespace,
+			'/comments/(?P<comment_id>\d+)/likes/new',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'new_like' ),
+					'permission_callback' => array( $this, 'permission_callback' ),
+					'args'                => array(
+						'_locale' => array(
+							'required' => false,
+							'type'     => 'string',
+						),
+					),
+				),
+			)
+		);
+
+		// Endpoint for deleting the current user's like.
+		register_rest_route(
+			$this->namespace,
+			'/comments/(?P<comment_id>\d+)/likes/mine/delete',
+			array(
+				array(
+					'methods'             => \WP_REST_Server::CREATABLE,
+					'callback'            => array( $this, 'delete_like' ),
+					'permission_callback' => array( $this, 'permission_callback' ),
+				),
+			)
+		);
+	}
+
+	/**
+	 * Permission callback.
+	 *
+	 * @return bool
+	 */
+	public function permission_callback(): bool {
+		return current_user_can( 'moderate_comments' );
```

Comment:
Is this the correct capability? 🤔 According to Core docs, Authors and Contributors don't have access to this capability.

The logic in WPCOM looks more complicated (checks the site visibility, if the user can view the post, etc). 

I'll test it tomorrow to see how it behaves with those roles.

Reply:
All of those wpcom checks do still happen, they just happen when this API calls that API. Arguably you could bring them here to fail faster, but then there's more code to keep in sync.

Reply:
Testing still seems like a good idea though 👍

Reply:
Users who can moderate comments will be the ones who visit that comments page most often.